sql/src/main/java/org/elasticsearch/action/bulk/BulkShardProcessor.java

```diff
@@ -343,8 +355,12 @@ public void onSuccess(@Nullable Void result) {
                         // add pending requests for created indices
                         ShardId shardId = shardId(pendingRequest.indexName, pendingRequest.item.id(), pendingRequest.routing);
                         if (shardId == null) {
-                            // seems like index is deleted meanwhile, mark item as failed and remove pending
-                            indicesDeleted.add(pendingRequest.indexName);
+                            // seems like index is deleted meanwhile
+                            // only mark item as failed if the index IS NOT closed
+                            if (indexClosedAndPrintLog(pendingRequest.indexName) == false) {
```

Comment:
Is it okay that the `indexClosedAndPrintLog` call here can also decrement the `pending` counter?